app/screens/create_direct_message/create_direct_message.tsx

```diff
@@ -87,56 +57,31 @@ export default function CreateDirectMessage({
     currentUserId,
     restrictDirectMessage,
     teammateNameDisplay,
-    tutorialWatched,
 }: Props) {
     const serverUrl = useServerUrl();
     const theme = useTheme();
-    const style = getStyleFromTheme(theme);
     const intl = useIntl();
-    const {formatMessage} = intl;
 
     const searchTimeoutId = useRef<NodeJS.Timeout | null>(null);
-    const next = useRef(true);
     const page = useRef(-1);
-    const mounted = useRef(false);
 
-    const [profiles, dispatchProfiles] = useReducer(reduceProfiles, []);
     const [searchResults, setSearchResults] = useState<UserProfile[]>([]);
     const [loading, setLoading] = useState(false);
     const [term, setTerm] = useState('');
-    const [startingConversation, setStartingConversation] = useState(false);
     const [selectedIds, setSelectedIds] = useState<{[id: string]: UserProfile}>({});
-    const selectedCount = Object.keys(selectedIds).length;
-
-    const isSearch = Boolean(term);
-
-    const loadedProfiles = ({users}: {users?: UserProfile[]}) => {
-        if (mounted.current) {
-            if (users && !users.length) {
-                next.current = false;
-            }
-
-            page.current += 1;
-            setLoading(false);
-            dispatchProfiles({type: 'add', values: users});
-        }
-    };
+    const selectedCount = useMemo(() => Object.keys(selectedIds).length, [selectedIds]);
```

Comment:
This is a number (base type) so no need for the use memo in the case of "new reference on every render".
The other reason to use use memo is the "expensive operation on every render". Do you consider the `Object.keys(selectedIds).length` is heavy enough to add the use of `useMemo`? 0/5.